release.py

```diff
@@ -0,0 +1,498 @@
+            )
+        countdown('Exiting and opening the charm pins PR in: ', 3)
+        webbrowser.open(pr.html_url)
+        exit(1)
+
+
+def release(owner: str, repo_name: str, branch: str):
+    """Create a release, update changelog, and create a PR for the release."""
+    org = gh_client.get_organization(owner)
+    repo = org.get_repo(repo_name)
+
+    check_update_charm_pins_prs(repo)
+
+    tag = get_new_version(owner, repo, branch)
+
+    release = create_draft_release(repo, tag, branch)
+    if not release:
+        logger.error('Failed to create draft release.')
+        exit(1)
+    logger.info(f'Draft release created: {release.html_url}.')
+
+    categories, full_changelog = parse_release_notes(release.body)
+    notes = format_notes(categories, full_changelog)
+    print_notes(notes)
+    title, summary = get_title_and_summary(release.title)
+    if not title:
+        title = tag
+    notes = f'{summary}\n{notes}'
+    update_release(release, title, notes)
+
+    changes = format_changes(categories, tag)
+    update_changes_file(changes, 'CHANGES.md')
+
+    update_versions_for_release(tag)
+
+    new_branch = f'release-prep-{tag}'
+    subprocess.run(['/usr/bin/git', 'checkout', '-b', new_branch], check=True)
+    changed_files = ['CHANGES.md', *VERSION_FILES]
+    for file in changed_files:
+        subprocess.run(['/usr/bin/git', 'add', file], check=True)
+    subprocess.run(['/usr/bin/git', 'commit', '-m', f'chore: prepare release {tag}'], check=True)
+    subprocess.run(['/usr/bin/git', 'push', 'origin', new_branch], check=True)
+    pr = repo.create_pull(
+        title=f'Release Preparation: {tag}',
+        body=f'This PR prepares the release of version {tag}.',
+        head=f'{gh_client.get_user().login}:{new_branch}',  # "your_username:new_branch"
+        base=branch,
+    )
+    logger.info(f'Created PR: {pr.html_url}')
+
+
+def post_release(owner: str, repo_name: str, branch: str):
+    """Post-release actions: update version files and create a PR."""
+    subprocess.run(['/usr/bin/git', 'fetch', 'upstream'], check=True)
+    subprocess.run(['/usr/bin/git', 'checkout', branch], check=True)
+    subprocess.run(['/usr/bin/git', 'merge', f'upstream/{branch}'], check=True)
+
+    org = gh_client.get_organization(owner)
+    repo = org.get_repo(repo_name)
+
+    update_versions_for_post_release()
+
+    new_branch = 'post-release'
+    subprocess.run(['/usr/bin/git', 'checkout', '-b', new_branch], check=True)
+    for file in VERSION_FILES:
+        subprocess.run(['/usr/bin/git', 'add', file], check=True)
+    subprocess.run(['/usr/bin/git', 'commit', '-m', 'chore: post release'], check=True)
+    subprocess.run(['/usr/bin/git', 'push', 'origin', new_branch], check=True)
+    pr = repo.create_pull(
+        title='Post Release',
+        body='This PR updates the version files after the release.',
+        head=f'{gh_client.get_user().login}:{new_branch}',  # "your_username:new_branch"
```

Comment:
Corner cases:
- `post-release` branch exists locally
- `post-release` branch exists on GitHub but not locally

Perhaps the script could take a stance here--either fail, on `check=True` (current behaviour) or overwrite local and remote branches with this name, or generate a random name, like `f"post-release-{some_hex}"` (maybe secrets.token_hex() or in fact current git commit)